src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -609,24 +462,10 @@ public static float SumAbs(ReadOnlySpan<float> source)
         public static float SumAbs(float mean, ReadOnlySpan<float> source)
         {
             Contracts.AssertNonEmpty(source);
+            if (mean == 0)
+                return TensorPrimitives.SumOfMagnitudes(source);
 
-            if (source.Length < MinInputSize || !Sse.IsSupported)
-            {
-                float sum = 0;
-                for (int i = 0; i < source.Length; i++)
-                {
-                    sum += Math.Abs(source[i] - mean);
-                }
-                return sum;
-            }
-            else if (Avx.IsSupported)
-            {
-                return (mean == 0) ? AvxIntrinsics.SumAbsU(source) : AvxIntrinsics.SumAbsDiffU(mean, source);
-            }
-            else
-            {
-                return (mean == 0) ? SseIntrinsics.SumAbsU(source) : SseIntrinsics.SumAbsDiffU(mean, source);
-            }
+            return TensorPrimitives.SumOfMagnitudes(source) - (mean * source.Length);
```

Comment:
```C#
float sum = TensorPrimitives.SumOfMagnitudes(source);
if (mean != 0)
{
    sum -= mean * source.Length;
}
return sum;
```
though is the extra branch cheaper than doing the single multiplication and just letting that be `- 0`?

Reply:
I'm not sure actually. Let me run some tests. I just copied the same logic from what we had before.